src/containers/main/NewTestGenerator.js

```diff
@@ -0,0 +1,129 @@
+import React, {Component} from "react";
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import StaticBlock from "../../components/blocks/StaticBlock";
+import DropContainer from "../../containers/blocks/DropBlock";
+import UserBlocksContainer from "../../containers/blocks/UserBlock";
+import verifyDependencies from "../../functions/verifyDependencies";
+import OptionsBox from "../../containers/popups/OptionsBox";
+import EditBox from "../../containers/popups/EditBox";
+import AddURLBlock from "../../containers/blocks/AddURLBlock";
+import TestNameBlock from "../../containers/popups/TestName"
+import {checkIfTestAlreadyExists} from "../../functions/checkIfTestExists"
+import ProjectInfo from "../../containers/test/ProjectInfo";
+import {hideEditBox, hideOptionsBox, hideUrlInput, hideTestNameInput, addTestToProject, clearTests, clearUrlsList} from "../../actions/";
+
+class NewTestGenerator extends Component {
+
+  componentDidMount() {
+    this.props.hideOptionsBox();
+    this.props.hideEditBox();
+  }
+
+  componentDidUpdate() {
+    verifyDependencies(this.props.test);
+  }
+
+  hideBoxes(ev) {
+    if(ev.target.classList.contains("tests-wrapper")
+    || ev.target.classList.contains("test-container") 
+    || ev.target.classList.contains("comparators-list-container") 
+    || ev.target.classList.contains("modifiers-list-container") 
+    || ev.target.classList.contains("collectors-list-container")
+    || ev.target.classList.contains("tests")
+    ) {
+      this.props.hideEditBox();
+      this.props.hideOptionsBox();
+      this.props.hideUrlInput();
+      this.props.hideTestNameInput();
+      document.querySelectorAll(".block").forEach((block) => {
+        block.classList.remove("block-hidden");
+      });
+    }
+  }
+
+  handleNewTest() {
+    if(this.props.test.length > 0 && this.props.testName.name && this.props.urls.length > 0) {
+      if(!checkIfTestAlreadyExists(this.props.testName)) {
+        this.props.addTestToProject(this.props.test, this.props.urls, this.props.testName);
+        this.props.clearTests();
+        this.props.hideUrlInput();
+        this.props.clearUrlsList();
+        this.props.hideTestNameInput();
+        document.querySelectorAll(".block").forEach((block) => {
+          block.classList.remove("block-hidden");
+        });
+      } else {
+        alert("Test with that name already exists!");
+      }
+    }
+  }
+
+  handleKeyboardShortcuts(ev) {
+    if(ev.key === "Enter") {
+      if(typeof document.getElementsByClassName("edit-box")[0] !== "undefined") {
```

Comment:
Can this be achieved using react state instead?

Reply:
Done.